bitcoin/src/address.rs

```diff
@@ -1054,13 +1084,16 @@ impl Address<NetworkUnchecked> {
     /// For details about this mechanism, see section [*Parsing addresses*](Address#parsing-addresses)
     /// on [`Address`].
     #[inline]
-    pub fn assume_checked(self) -> Address { Address::new(self.network, self.payload) }
+    pub fn assume_checked(self) -> Address {
+        let (network, payload) = self.into_parts();
+        Address::new(network, payload)
+    }
```

Comment:
In c15b07e622df49127a38b72b005934ef8c158d00:

If we're using unsafe code for the reference version, we could use unsafe code and just transmute here. But this is fine, it's only one extra line of code and is safe.

Reply:
I'm pretty sure it'll compile down to the same code and I don't want to add more `unsafe` than required to achieve what we need.